org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/AnalyzeProjectJob.java

```diff
@@ -244,18 +325,23 @@ private void handleLinkedFiles(Collection<File> tmpToDelete, final File baseDir)
     }
   }
 
-  public Map<IResource, List<Issue>> run(final Properties props, final SonarRunnerFacade runner, final IProgressMonitor monitor) {
-    SonarLintCorePlugin.getDefault().debug("Start analysis with args:\n" + propsToString(props));
+  public Map<IResource, List<Issue>> run(final AnalysisConfiguration config, final SonarLintClientFacade sonarlint, final IProgressMonitor monitor) {
+    SonarLintCorePlugin.getDefault().debug("Start analysis with configuration:\n" + config.toString());
     Thread.UncaughtExceptionHandler h = exceptionHandler();
     final Map<IResource, List<Issue>> issuesPerResource = new HashMap<>();
     Thread t = new Thread("SonarLint analysis") {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this anonymous class number of lines from 24 to at most 20, or make it a named class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1188)